### PR TITLE
x86: minowboard: Enable userspace mode

### PR DESCRIPTION
**Pull request by AdithyaBaglody**

### CODE REVIEW

File: include/arch/x86/linker.ld [hunk 1]
```diff
@@ -159,6 +159,9 @@ SECTIONS
 #ifdef CONFIG_APPLICATION_MEMORY
 	SECTION_DATA_PROLOGUE(_APP_DATA_SECTION_NAME, (OPTIONAL),)
 	{
+#ifndef CONFIG_XIP
```
Comment (andrewboie) [inline include/arch/x86/linker.ld]: why was this change made? should be in its own commit

**Review by tyrantlizard77 — APPROVED** [review]
